src/login/logind-session.c

```diff
@@ -179,6 +179,23 @@ int session_set_leader(Session *s, pid_t pid) {
         return 1;
 }
 
+int session_invalidate_leader(Session *s) {
+        assert(s);
+
+        if (s->leader <= 0)
```

Comment:
Maybe use `pid_is_valid()` here?

Reply:
That's probably a bit nicer. Anyway the result is the same.